reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/driver/YarnContainerManager.java

```diff
@@ -659,22 +661,25 @@ private String getChangeLogLocation() {
 
   private void logContainerAddition(final String containerId) {
     final String entry = ADD_FLAG + containerId + System.lineSeparator();
-    try {
-      writeToEvaluatorLog(entry);
-    } catch (final IOException e) {
-      LOG.log(Level.WARNING, "Unable to log the addition of container [" + containerId +
-          "] to the container log. Driver restart won't work properly.", e);
-    }
+    logContainerChange(entry);
   }
 
   private void logContainerRemoval(final String containerId) {
     final String entry = REMOVE_FLAG + containerId + System.lineSeparator();
-    try {
-      writeToEvaluatorLog(entry);
-    } catch (final IOException e) {
-      LOG.log(Level.WARNING, "Unable to log the removal of container [" + containerId +
-          "] to the container log. Driver restart won't work properly.", e);
-    }
+    logContainerChange(entry);
   }
 
+  private void logContainerChange(final String entry)
+  {
+    synchronized (this){
```

Comment:
It may be better to remove the synchronized blocks at L615 and L674 and instead make `writeToEvaluatorLog` a synchronized method.